app/routes/jokes.tsx

```diff
@@ -1,33 +1,30 @@
 import type { LinksFunction, LoaderFunction } from "@remix-run/node";
-import type { Joke } from "@prisma/client";
 import { json } from "@remix-run/node";
 import { Link, Outlet, useLoaderData } from "@remix-run/react";
+import type { UnpackResult } from "domain-functions";
+import { listUserJokes } from "~/domains/jokes";
+import { getUserId, logout } from "~/utils/session.server";
 
-import { db } from "~/utils/db.server";
 import stylesUrl from "~/styles/jokes.css";
 
 export const links: LinksFunction = () => {
   return [{ rel: "stylesheet", href: stylesUrl }];
 };
 
-type LoaderData = {
-  jokeListItems: Array<{ id: string; name: string }>;
-};
+type LoaderData = UnpackResult<typeof listUserJokes>;
 
-export const loader: LoaderFunction = async () => {
-  const data: LoaderData = {
-    jokeListItems: await db.joke.findMany({
-      take: 5,
-      select: { id: true, name: true },
-      orderBy: { createdAt: "desc" },
-    }),
-  };
-  return json(data);
+export const loader: LoaderFunction = async ({ request }) => {
+  const result = await listUserJokes(null, await getUserId(request));
+  if (result.errors.length) throw logout(request);
```

Comment:
wouldn't it be easier to logout on every DF failure?
```suggestion
  if (!result.success) throw logout(request);
```

Reply:
Can't accept, it breaks the application